src/components/Tabs/PlotTab.tsx

```diff
@@ -26,36 +27,50 @@ const TrackSearch = styled(FlexRowAlignCenter)`
 `;
 
 type PlotTabProps = {
-  findTrackInputText: string;
-  setFindTrackInputText: (text: string) => void;
-  findTrack: (trackId: number, seekToFrame?: boolean) => void;
-  setFrame: (frame: number) => Promise<void>;
-  currentFrame: number;
-  dataset: Dataset | null;
-  featureKey: string | null;
-  selectedTrack: Track | null;
   disabled: boolean;
 };
 
 export default function PlotTab(props: PlotTabProps): ReactElement {
-  const [isLoading, setIsLoading] = useState(false);
-  const pendingFrame = useRef<number>(props.currentFrame);
+  const store = useViewerStateStore(
+    useShallow((state) => ({
+      currentFrame: state.currentFrame,
+      pendingFrame: state.pendingFrame,
+      dataset: state.dataset,
+      featureKey: state.featureKey,
+      selectedTrack: state.track,
+      setFrame: state.setFrame,
+      setTrack: state.setTrack,
+    }))
+  );
+
+  // Sync track searchbox with selected track
+  useEffect(() => {
+    const unsubscribe = useViewerStateStore.subscribe(
+      (state) => [state.track],
+      ([track]) => {
+        if (track) {
+          setFindTrackInput(track.trackId.toString());
+        } else {
+          setFindTrackInput("");
+        }
+      }
+    );
+    return unsubscribe;
+  }, []);
+
+  const [findTrackInput, setFindTrackInput] = useState("");
+  const isLoading = store.currentFrame !== store.pendingFrame;
```

Comment:
bit confusing that this effect definition uses a state setter which is defined after it

Reply:
Oh, good point. I'll move that.